From bb5704207f57167e46bb0f81994b813fcac4f035 Mon Sep 17 00:00:00 2001
From: Markus Stockhausen <markus.stockhausen@gmx.de>
Date: Tue, 13 Aug 2024 09:46:22 -0400
Subject: [PATCH] realtek net: phy: EEE support for rtl838x

* rename the target to realtek
* add refactored DSA driver
* add latest gpio driver
* lots of arch cleanups
* new irq driver
* additional boards

Submitted-by: Bert Vermeulen <bert@biot.com>
Submitted-by: Birger Koblitz <mail@birger-koblitz.de>
Submitted-by: Sander Vanheule <sander@svanheule.net>
Submitted-by: B Mork <bjorn@mork.no>
Submitted-by: John Crispin <john@phrozen.org>
---
 drivers/net/phy/phylink.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

--- a/drivers/net/phy/phylink.c
+++ b/drivers/net/phy/phylink.c
@@ -2484,6 +2484,11 @@ int phylink_ethtool_ksettings_set(struct
 		 *   the presence of a PHY, this should not be changed as that
 		 *   should be determined from the media side advertisement.
 		 */
+		if (pl->phydev->drv->get_port && pl->phydev->drv->set_port) {
+			if(pl->phydev->drv->get_port(pl->phydev) != kset->base.port) {
+				pl->phydev->drv->set_port(pl->phydev, kset->base.port);
+			}
+		}
 		return phy_ethtool_ksettings_set(pl->phydev, &phy_kset);
 	}
 
@@ -2786,8 +2791,11 @@ int phylink_ethtool_get_eee(struct phyli
 
 	ASSERT_RTNL();
 
-	if (pl->phydev)
+	if (pl->phydev) {
+		if (pl->phydev->drv->get_eee)
+			return pl->phydev->drv->get_eee(pl->phydev, eee);
 		ret = phy_ethtool_get_eee(pl->phydev, eee);
+	}
 
 	return ret;
 }
@@ -2804,8 +2812,11 @@ int phylink_ethtool_set_eee(struct phyli
 
 	ASSERT_RTNL();
 
-	if (pl->phydev)
+	if (pl->phydev) {
+		if (pl->phydev->drv->set_eee)
+			return pl->phydev->drv->set_eee(pl->phydev, eee);
 		ret = phy_ethtool_set_eee(pl->phydev, eee);
+	}
 
 	return ret;
 }
